core/src/main/scala/org/apache/pekko/persistence/r2dbc/ConnectionFactoryProvider.scala

```diff
@@ -40,6 +44,14 @@ object ConnectionFactoryProvider extends ExtensionId[ConnectionFactoryProvider]
 
   // Java API
   def get(system: ActorSystem[_]): ConnectionFactoryProvider = apply(system)
+
+  trait ConnectionFactoryOptionsCustomizer {
+    def apply(options: ConnectionFactoryOptions, config: Config): ConnectionFactoryOptions
```

Comment:
Would it be better if argument and return type were `ConnectionFactoryOptions.Builder`?
```scala
def apply(builder: ConnectionFactoryOptions.Builder, config: Config): ConnectionFactoryOptions.Builder
```
Because for any mutation the `options` here would need to be converted back to builder again anyways.

Reply:
Yes I can change that. @pjfanning WDYT?

Reply:
seems good to me

Reply:
Fixed